_sources/MoreAboutIteration/SentinelValuesAndValidation.rst

```diff
@@ -112,13 +112,13 @@ When you run the following code, try typing something other than Y or N to see h
 
     def get_yes_or_no(message):
         valid_input = False
+        answer = input(message)
         while not valid_input:
-            answer = input(message)
             answer = answer.upper() # convert to upper case
             if answer == 'Y' or answer == 'N':
                 valid_input = True
             else:
-                print('Please enter Y for yes or N for no.')
+                answer = input('Please enter Y for yes or N for no. \n' + message)
```

Comment:
Both the old and the new version look right to me.  What is the glitch??

Reply:
Before it would print the 'Please enter Y for yes or N for no' in the console, so the user couldn't see it happening in real-time until they inserted Y or N. Only then would they see multiple lines of 'Please enter Y for yes or N for no'. 
We made it pop up in the box input-prompt so that if the user types something else, it will give them directions accordingly.